src/main/java/org/mule/extension/http/api/HttpRequestAttributesBuilder.java

```diff
@@ -139,7 +142,36 @@ public HttpRequestAttributes build() {
     requireNonNull(requestUri, "Request URI cannot be null.");
     requireNonNull(localAddress, "Local address cannot be null.");
     requireNonNull(remoteAddress, "Remote address cannot be null.");
-    return new HttpRequestAttributes(headers, listenerPath, relativePath, version, scheme, method, requestPath, requestUri,
+    this.proxyRequestPath = resolveProxyRequestPath();
+    return new HttpRequestAttributes(headers, listenerPath, relativePath, proxyRequestPath, version, scheme, method, requestPath,
+                                     requestUri,
                                      queryString, queryParams, uriParams, localAddress, remoteAddress, clientCertificate);
   }
+
+  private String resolveProxyRequestPath() {
+    byte[] listenerPathBytes = listenerPath.getBytes();
+    byte[] requestPathBytes = requestPath.getBytes();
+    int listenerPathIndex = 0;
```

Comment:
I think these two vars would make more sense as `listenerCurrentSlashIndex` or something that relates the index to the fact that it's pointing to a slash position